File: deck/base/code/native/rust/runtime/regex.rs
mod regex {
    pub fn matches(pattern: String, text: String) -> bool {
        ::regex::Regex::new(&pattern).map(|r| r.is_match(&text)).unwrap_or(false)
    }
    pub fn replace(pattern: String, text: String, replacement: String) -> String {
        ::regex::Regex::new(&pattern).map(|r| r.replace_all(&text, replacement.as_str()).to_string()).unwrap_or(text)
    }
    pub fn find(pattern: String, text: String) -> String {
        ::regex::Regex::new(&pattern).ok().and_then(|r| r.find(&text).map(|m| m.as_str().to_string())).unwrap_or_default()
    }
}

File: deck/base/code/native/browser/runtime/regex.ts
// Regex shim over the host RegExp. Wraps construction so the native module can dock a uniform `regex` namespace.
const regex = {
  matches: (pattern: string, text: string): boolean =>
    new RegExp(pattern).test(text),
  replace: (
    pattern: string,
    text: string,
    replacement: string,
  ): string => text.replace(new RegExp(pattern, 'g'), replacement),
  find: (pattern: string, text: string): string => {
    const m = text.match(new RegExp(pattern))
    return m ? m[0] : ''
  },
}
